Implementation of Use case Reset Password #5

sebastien-deren opened the pull request:
Creation of our mailsender
Creation of our reset password form,
our use case is implemented, and relatively SOLID
We'll need to implement a token hashed properly (probably with our secret key) to verify the mail properly

sebastien-deren committed:
creation of our controller &form"
Creation of our mailsender
Creation of our reset password form,
our use case is implemented, and relatively SOLID
We'll need to implement a token hashed properly (probably with our secret key) to verify the mail properly

sebastien-deren committed:
SOLID ify the use case
Adding better (but still eye damagaging) view

Lionelsan95 commented on src/Controller/ResetPasswordController.php:
```diff
@@ -0,0 +1,67 @@
+        if ($form->isSubmitted() && $form->isValid()) {
+            $mail = $form->get('mail')->getData();
+            try {
+                $this->service->ResetPasswordRequest($mail, 'app_reset_password_mail');
```
@sebastien-deren si ça se passe bien on est pas redirigé une page différent de reset_password/index ?

sebastien-deren replied:
Corrigé, création de template twig pour l'email envoyé et pour changer de mot de passe.

Lionelsan95 commented on src/Controller/ResetPasswordController.php:
```diff
@@ -0,0 +1,67 @@
+    #[Route('/email/{id}/{token}', name: 'app_reset_password_mail')]
+    public function resetFromMail(Request $request, User $user, string $token): Response
+    {
+        if (!hash_equals($token, hash('md5', $user->getId() . $user->getUsername()))) {
```
@sebastien-deren tu peux créer une méthode dans la classe User qui te renvoie le hash md5

return hash('md5', $user->getId() . $user->getUsername());

ainsi tu ferais juste : 
has_equals($token,$user->getHash()); 

ou un autre nom

sebastien-deren replied:
Corrigé! Création d'une méthode dans user.

Lionelsan95 commented on src/Controller/ResetPasswordController.php:
```diff
@@ -0,0 +1,67 @@
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->service->upgradePassword($user, $form->get('plain_password')->getData());
+            dd($user);
```
@sebastien-deren à supprimer ultérieurement je pense

sebastien-deren replied:
corrigé!

Lionelsan95 commented on src/Service/Interfaces/MailPasswordInterface.php:
```diff
@@ -0,0 +1,11 @@
+
+use Symfony\Component\Mime\Address;
+
+interface MailPasswordInterface
```
@sebastien-deren nécessaire ?

sebastien-deren replied:
corrigé!

Lionelsan95 reviewed (commented):
waiting for update and/or clarifiations to test and valide

sebastien-deren committed:
taking into account mentor's review
adding a method in User to get a hash.
adding new templates

sebastien-deren committed:
Using bootstrap to make a proper ResetPw page
Modification of the resetPassword, asking username in place of an email as seen in specs

Lionelsan95 commented:
Après tests, ça update bien le passage en bdd. 

remarques : 
- après que l'action ai été effectuée, on a pas de moyens de savoir (pas de redirection ou message)

Lionelsan95 commented:
Lors du reset et de la création de compte, il faut toujours un double champs pour confirmer le mot de passe choisi. ça reste ultra import ce champs et on peut faire des erreurs.

sebastien-deren committed:
Adding a second password field to ensure typo
Redirect to the login page when the password as been updated
(with an alert password updated!)

sebastien-deren committed:
correcting conflicting files

sebastien-deren committed:
correction after merging develop
correcting asserts
correcting routing
testing done and working
